File: 2020/1.py
from framework import *

d = set([*map(int, read_input_lines())])

for x in d:
    if 2020 - x in d:
        print((2020 - x) * x)
        break

p2 = None

for x in d:
    for y in d:
        if 2020 - x - y in d:
            p2 = (2020 - x - y) * x * y
            break
print(p2)

File: 2022/1.hs
{-# OPTIONS_GHC -Wno-incomplete-patterns #-}
import Framework
import Data.List
import Data.Maybe

splitElves l = case break null l of
    (s, []) -> [s]
    (s, "" : rest) -> s : splitElves rest

sumCalories = map (sum . map read)

day1 = maximum . sumCalories . splitElves . lines
day2 = sum . take 3 . reverse . sort . sumCalories . splitElves . lines
